File: app/src/main/java/at/shockbytes/coins/storage/realm/model/ShockConfig.java
package at.shockbytes.coins.storage.realm.model;

import io.realm.RealmObject;


public class ShockConfig extends RealmObject {

    private long lastPrimaryKey;

    public ShockConfig() {
        this(0);
    }

    public ShockConfig(long lastPrimaryKey) {
        this.lastPrimaryKey = lastPrimaryKey;
    }

    public long getLastPrimaryKey() {
        long key = lastPrimaryKey;
        lastPrimaryKey ++;
        return key;
    }

}

File: app/src/main/java/at/shockbytes/coins/storage/realm/model/ShockConfig.kt
package at.shockbytes.coins.storage.realm.model

import io.realm.RealmObject


open class ShockConfig(private var lastPrimaryKey: Long = 0) : RealmObject() {

    fun getLastPrimaryKey(): Long {
        val key = lastPrimaryKey
        lastPrimaryKey++
        return key
    }

}
